channels/chan_pjsip.c

```diff
@@ -850,7 +855,68 @@ static struct ast_frame *chan_pjsip_read_stream(struct ast_channel *ast)
 	}
 
 	callback_state = AST_VECTOR_GET_ADDR(&session->active_media_state->read_callbacks, fdno);
-	f = callback_state->read_callback(session, callback_state->session);
+
+	if (ast_avoxi_purge_packets(ast)) {
+		ast_log(LOG_DEBUG, "AVOXI: chan_pjsip_read_stream(): Purge flag is set Ch=%s ChSt: %d",
+				ast_channel_name(ast), ast_channel_state(ast));
+
+		int drop_count = 0;
+		long long last_usec = 0;
+		const int max_packets_to_drop = 150;
+		for(int i = 0; i < max_packets_to_drop; i++) {
+			f = callback_state->read_callback(session, callback_state->session);
+			if (!f) {
+				break;
+			}
+			if (f->frametype != AST_FRAME_VOICE) {
+				continue;
```

Comment:
Missing ast_frfree(f); ?

Reply:
good point, we are freeing the processed frames below, looks like we are leaking frames if they are not of `AST_FRAME_VOICE` type. This has already been in production for ~a month, I am going to leave it there to be on the safe side and create a ticket for this.